### PR TITLE
1.2 - GEOMESA-1358 Configurable precision for XZ indices

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: docs/user/data_management.rst [hunk 1]
```diff
@@ -186,6 +186,29 @@ user data using the hint ``geomesa.z3.interval``:
...
 See below for alternate ways to set the user data.
 
+.. _customizing_xz_index:
+
+Customizing the XZ-Index
+------------------------
+
+GeoMesa uses an extended z-curve index for storing geometries with extents. The index can be customized
+by specifying the resolution level used to store geometries. By default, the resolution level is 12. If
+you have very small geometries, or very large geometries, you may want to tweak this value.
```
Comment (cne1x) [inline docs/user/data_management.rst]: ... in which direction?

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/z2/XZ2WritableIndex.scala [hunk 2]
```diff
@@ -20,29 +20,31 @@ import org.locationtech.geomesa.accumulo.AccumuloVersion
...
 import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
 import org.opengis.feature.simple.SimpleFeatureType
 
 trait XZ2WritableIndex extends AccumuloWritableIndex {
 
   import AccumuloWritableIndex.{DefaultNumSplits, DefaultSplitArrays}
-  import XZ2Index.SFC
 
   override def writer(sft: SimpleFeatureType, ops: AccumuloDataStore): (WritableFeature) => Seq[Mutation] = {
+    val sfc = XZ2SFC(sft.getXZPrecision)
```
Comment (cne1x) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/z2/XZ2WritableIndex.scala]: Could we still have a lazy variable at the trait level that gets shared among these routines?

Comment (cne1x) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/z2/XZ2WritableIndex.scala]: Never mind.  :(

Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/z2/XZ2WritableIndex.scala]: the sfcs are cached in the apply method, so it should mostly be re-using the same instance object.

File: docs/user/data_management.rst (hunk 1 above)
Comment (elahrvivaz) [inline docs/user/data_management.rst]: udpated

**Pull request merged**